File: src/Telebox/Data/Connection.cs
namespace Telebox.Data;

public class Connection
{
    public int Id { get; set; }

    public string Name { get; set; } = default!;

    public string Username { get; set; } = default!;

    public string Password { get; set; } = default!;

    public string ServerUrl { get; set; } = default!;
    
    public DateTime CreatedAt { get; set; }
}

File: src/Telebox/ClientApp/src/models/Connection.ts
export interface IConnection {
  id: number;
  name: string;
  username: string;
  password: string;
  serverUrl: string;
  createdAt: string;
}

export interface ICreateConnectionRequest {
  name: string;
  username: string;
  password: string;
  serverUrl: string;
}

export const EmptyCreateConnectionRequest: ICreateConnectionRequest = {
  name: "",
  username: "",
  password: "",
  serverUrl: "",
};
